bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/discovery/HueBridgeMDNSDiscoveryParticipant.java

```diff
@@ -104,29 +115,66 @@ public String getServiceType() {
     public @Nullable DiscoveryResult createResult(ServiceInfo service) {
         if (isAutoDiscoveryEnabled) {
             ThingUID uid = getThingUID(service);
-            if (uid != null) {
+            if (Objects.nonNull(uid)) {
                 String host = service.getHostAddresses()[0];
-                String id = service.getPropertyString(MDNS_PROPERTY_BRIDGE_ID);
-                String friendlyName = String.format("%s (%s)", service.getName(), host);
-                return DiscoveryResultBuilder.create(uid) //
-                        .withProperties(Map.of( //
-                                HOST, host, //
-                                Thing.PROPERTY_MODEL_ID, service.getPropertyString(MDNS_PROPERTY_MODEL_ID), //
-                                Thing.PROPERTY_SERIAL_NUMBER, id.toLowerCase())) //
-                        .withLabel(friendlyName) //
+                String serial = service.getPropertyString(MDNS_PROPERTY_BRIDGE_ID);
+                String label = String.format("%s (%s)", service.getName(), host);
+                String legacyThingUID = null;
+
+                if (uid.getAsString().startsWith(HueBindingConstants.CLIP2_THING_TYPE_PREFIX)) {
```

Comment:
I don't think so. Yes, we need to compare two Thing**Type**UID's. But `uid.getId()` does not return a ThingTypeUID, it actually returns the string value of the third segment of a ThingUID.

Reply:
Ah, you're right. So I guess this would work:
```java
if (new ThingUID(HueBindingConstants.THING_TYPE_CLIP2, uid.getId()).equals(uid)) {
```

It's a little clumsy, yes. Maybe there is a shorter path.

This also seems a little hack'ish in combination with `String.startsWith`:
```java
public static final String CLIP2_THING_TYPE_PREFIX = "hue:clip2:";
```

But I guess another option would be replacing hardcoded literals and removing duplication:
```java
    private static final String CLIP2_BRIDGE_TYPE = "clip2";

    public static final ThingTypeUID THING_TYPE_CLIP2 = new ThingTypeUID(BINDING_ID, CLIP2_BRIDGE_TYPE);
    public static final String CLIP2_THING_TYPE_PREFIX = BINDING_ID + AbstractUID.SEPARATOR + CLIP2_BRIDGE_TYPE + AbstractUID.SEPARATOR;
```

By using the `ThingUID` class we would not have to make assumptions about how the segments are organized and used to build the UID, though.

Reply:
Ok. Done.